### PR TITLE
Config Enhancements 

### PR DESCRIPTION
**Pull request by vaibhavchellani**
- add version command 
- makes `heimdall-config` toml instead of json
- removed consensusPubKey etc
- adds build  and ld flags

### CODE REVIEW

**Comment by vaibhavchellani** [comment]
There is an issue reading the `toml` file, resolving, review after update please.

**Comment by vaibhavchellani** [comment]
Fixed!

File: cmd/heimdalld/main.go [hunk 1]
```diff
@@ -181,16 +168,9 @@ func InitCmd(ctx *server.Context, cdc *codec.Codec) *cobra.Command {
...
-			if err != nil {
-				return err
-			}
-
-			if err := common.WriteFileAtomic(filepath.Join(config.RootDir, "config/heimdall-config.json"), heimdallConfBytes, 0600); err != nil {
-				fmt.Println("Error writing heimdall-config", err)
-				return err
-			}
+			heimdallConf := helper.GetDefaultHeimdallConfig()
+			fmt.Println("pringitn heimdall-config", heimdallConf.MaxCheckpointLength)
```
Comment (mankenavenkatesh) [inline cmd/heimdalld/main.go]: spell check in log. 
Also can we use logger?

File: helper/config.go [hunk 2]
```diff
@@ -87,35 +88,37 @@ func init() {
...
+	ValidatorSetAddress  string `mapstructure:"validator_set_contract"`  // Validator Set contract address on bor chain
 
-	ChildBlockInterval uint64 `json:"childBlockInterval"` // Difference between header index of 2 child blocks submitted on main chain
+	ChildBlockInterval uint64 `mapstructure:"child_chain_block_interval"` // Difference between header index of 2 child blocks submitted on main chain
 
 	// config related to bridge
-	CheckpointerPollInterval int           `json:"checkpointerPollInterval"` // Poll interval for checkpointer service to send new checkpoints or missing ACK
-	SyncerPollInterval       int           `json:"syncerPollInterval"`       // Poll interval for syncher service to sync for changes on main chain
-	NoACKPollInterval        time.Duration `json:"noackPollInterval"`        // Poll interval for ack service to send no-ack in case of no checkpoints
+	CheckpointerPollInterval time.Duration `mapstructure:"checkpoint_poll_interval"` // Poll interval for checkpointer service to send new checkpoints or missing ACK
```
Comment (mankenavenkatesh) [inline helper/config.go]: Type Conflict.  In bridge checkpoint.go, 
startPolling(ctx context.Context, pollInterval int) 
expects int.

File: helper/config.go [hunk 3]
```diff
@@ -87,35 +88,37 @@ func init() {
...
+	SyncerPollInterval       time.Duration `mapstructure:"syncer_poll_interval"`     // Poll interval for syncher service to sync for changes on main chain
```
Comment (mankenavenkatesh) [inline helper/config.go]: Type Conflict. In bridge syncer.go, 
startPolling(ctx, helper.GetConfig().SyncerPollInterval)
expects int.

File: cmd/heimdalld/main.go (hunk 1 above)
Comment (vaibhavchellani) [reply cmd/heimdalld/main.go]: Sorry, wasn't supposed to push this!

**Comment by mankenavenkatesh** [comment]
LGTM

**Pull request merged**